### PR TITLE
HV-1831 experiments (6.2)

### PR DESCRIPTION
**Pull request by gbadner**
https://hibernate.atlassian.net/browse/HV-1831

Supersedes https://github.com/hibernate/hibernate-validator/pull/1169, which targets the incorrect branch.

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/engine/tracking/PredefinedScopeProcessedBeansTrackingStrategy.java [hunk 1]
```diff
@@ -160,38 +159,38 @@ private boolean determineTrackingRequired(Class<?> beanClass, Set<Class<?>> bean
...
-							for ( Type typeArgument :  parameterizedType.getActualTypeArguments() ) {
-								if ( typeArgument instanceof Class ) {
-									directCascadedBeanClasses.add( (Class<?>) typeArgument );
-								}
-								else {
-									throw new UnsupportedOperationException( "Only ParameterizedType values of type Class are supported" );
-								}
+			final BeanMetaData<?> beanMetaData = rawBeanMetaDataMap.get( beanClass );
+
+			if ( beanMetaData == null || !beanMetaData.hasCascadables() ) {
```
Comment (gbadner) [inline engine/src/main/java/org/hibernate/validator/internal/engine/tracking/PredefinedScopeProcessedBeansTrackingStrategy.java:164]: I was wondering how to deal with `beanMetaData == null`. I'm glad to see that spelled out here.

**Comment by gbadner** [comment]
@gsmet, I've read through your changes, and I understand why you made those changes. Looks good to me!

**Comment by jameswynn** [comment]
Has this stalled? Any chance we will see a resolution to this in a near release?

**Comment by gsmet** [comment]
I recreated the PR with a branch in my fork here: https://github.com/hibernate/hibernate-validator/pull/1288 .

**Comment by gsmet** [comment]
@jameswynn I don't have much time to spend on HV lately. If someone is interested in pursuing this in the meantime, it's very welcome.

**Comment by sjaakd** [comment]
@gsmet : we have similar issues with the current solution. I'm having troubles finding the latest status of this. Is somebody still working on this? Thanks..

**Pull request closed without merging**